lib/spack/spack/test/cmd/env.py

```diff
@@ -869,6 +870,102 @@ def test_env_with_included_config_file(mutable_mock_env_path, packages_file):
     assert any(x.satisfies("mpileaks@2.2") for x in e._get_environment_specs())
 
 
+def test_config_change_existing(mutable_mock_env_path, tmp_path, mock_packages, mutable_config):
```

Comment:
I'm doing unrelated work to polish / speed-up unit-tests. Am I correct that this test should have been marked as `only_clingo`, since the old concretizer cannot deal with `require:`?  Should we investigate why the test seems to be currently passing in CI?